sale_timesheet_analytic_propagation/models/hr_timesheet.py

```diff
@@ -0,0 +1,22 @@
+from odoo import api, fields, models
+
+
+class HrTimehseet(models.Model):
+    _inherit = "account.analytic.line"
+
+    account_id = fields.Many2one(
+        compute="_compute_account_id",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("task_id.analytic_account_id")
+    def _compute_account_id(self):
+        """Change the analytic account non billed lines."""
+        timesheet_ids = self.filtered(
+            lambda t: t.task_id.analytic_account_id
```

Comment:
I'm not sure it's optimal that way for multiple editing, maybe it's better to check `is_so_line_edited` first prior to tast_id unwrapping?

Reply:
Also, if I want to erase the account_id, it won't be possible due to `t.task_id.analytic_account_id` check? Please cover that case in tests explicitly

Reply:
I've open a [new PR](https://github.com/OCA/timesheet/pull/638) for this module.

Now `is_so_line_edited` is evaluated first as you suggested.

Regarding to delete account_id:
It is not possible for an "account.analytic.line" to not have an account_id as is required. Also that account_id has a ondelte restrict.

Thank you for your reviewing.